Fixed all previous issues

FedG-code opened the pull request:
- Titles are now fixed
- og tags are now in place with just an issue with spaces after punctuation in the description
- the list file now passes through the template, eventually I'll have to generate the file itself from all the blogs being turned into html
- final formatting issue: currently I have a div around the header, yet the div isn't taking the size of the header, this means I'm unable to add a bottom margin to the overall header meaning the text gets cut off at the top of the articles as shown below
![titlecutoff](https://user-images.githubusercontent.com/60428122/201708747-2df56195-b86e-4289-be5c-c1002125f0c6.jpg)

FedG-code committed:
Improvements

Better text formatting with margin usage. A header that at the moment doesn't take up the full page. A broken progress bar, I think my script might be in the wrong spot?

FedG-code committed:
Successful progress bar

Scrolling bar works but it's in the wrong place. The banner itself still isn't taking up the whole screen. I can't seem to find a way to isolate the margin to just the text I bring in.

FedG-code committed:
Updated images cleaner look

FedG-code committed:
Fixed header colour and allignment

FedG-code committed:
Header fixed

FedG-code committed:
Added list.html as a storage for the blog links

As long as with fixes to the image locations and the nav bar

FedG-code committed:
Completed all tasks

Titles are now accurate
Geo tags are now filled
The article list is now correctly passed through the template, eventually there will have to be code that makes the list file by itself with all the titles generated from the articles.

Final formatting problem is I can't seem to get the div around my banner to take the size of the banner, so that I can add a bottom margin and not cut off the top of the text in each page

robinglen commented on blog/converter.js:
```diff
@@ -1,19 +1,69 @@
+    const titleIdTag = titleFull.search('">');
+    titleFull = titleFull.substring(titleIdTag);
+    titleFull = titleFull.replace('">', '');
+    console.log(titleFull);
```
console

robinglen commented on blog/converter.js:
```diff
@@ -1,19 +1,69 @@
+    //pass the file through the converter
+    const newHTMLFile = convertToHTML.makeHtml(readableFile);
+    //separate title from body
+    const jsHeaderStart = '<h1';
```
all this method for finding tags could be much cleaner now youre using cheerio

robinglen commented on blog/converter.js:
```diff
@@ -1,19 +1,69 @@
+    titleFull = titleFull.replace('">', '');
+    console.log(titleFull);
+    //find the first image in the file
+    const firstImageLocationStart = newHTMLFile.indexOf('<img src=');
```
Same here, cheerio could be much cleaner way of going this

robinglen commented on blog/converter.js:
```diff
@@ -1,19 +1,69 @@
+    //take the first 100 words
+    rawDescription = rawDescription.split(' ').slice(0, 100).join(' ');
+    //since we have taken from multiple paragraphs there's instances of fullstops without a space after, let's fix that
+    rawDescription = rawDescription.replace('.', '. ');
```
I wonder if there is something that could clean up the code here better, like a HTML formatter or copy formatter... but we do it for the description why not the whole article?

robinglen commented on blog/microservices_solving-a-problem-like-routine_2020-update.html:
```diff
@@ -1,4 +1,67 @@
+      <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <title>Microservices: Solving a problem like routing — 2020 update</title>
```
This list has meta data from one article?

FedG-code replied:
I'm not sure what you're asking here

robinglen replied:
you can actually ignore this one now

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
```
If we are moving this into a .css we might want to consider doing this to the homepage and sharing

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,67 @@
+  gtag("js", new Date());
+  gtag("config", "G-BGXW90V3SE");
+</script>
+<meta property="og:title" content="{{title}}" />
```
put these with the meta tags

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,67 @@
+
+  <script>
+    // When the user scrolls the page, execute myFunction 
+    window.onscroll = function() {myFunction()};
```
myFunction and using vars inside it could be cleaned up

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,67 @@
+<div class="page-banner">
+  <nav>
+    
+    <a href="templatedlist.html">
```
lets use `-` to separate words

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,67 @@
+  <nav>
+    
+    <a href="templatedlist.html">
+      <img id="index-image" src="../img/gullwing-logo.png" alt ="Gullwing Logo">
```
could you think of a better icon here?

robinglen commented on blog/template.hbs:
```diff
@@ -0,0 +1,67 @@
+
+  <div class="header">
+    <div class="progress-container">
+      <div class="progress-bar" id="myBar"></div>
```
`myBar` could be better named

robinglen commented on blog/templatedlist.html:
```diff
@@ -0,0 +1,87 @@
+
+<body>
+
+  <script>
```
Do we need this here?

robinglen commented on blog/templatedlist.html:
```diff
@@ -0,0 +1,87 @@
+  gtag("js", new Date());
+  gtag("config", "G-BGXW90V3SE");
+</script>
+<meta property="og:title" content="Article List" />
```
these tags?

FedG-code committed:
Cheerio updated title and image grab

FedG-code committed:
Fixes

Fixed banner overlapping title.
Fixed banner bar not working after I renamed it.
Hardcoded og tags for article list page.
Fixed a previously unknown problem of image size.
